db/view/view.cc

```diff
@@ -2682,11 +2682,69 @@ static future<> flush_base(lw_shared_ptr<replica::column_family> base, abort_sou
     }).discard_result();
 }
 
+void view_builder::update_base_schema(const replica::column_family& base) {
+    auto base_schema = base.schema();
+    auto& views = base.views();
+    if (views.empty()) {
+        return;
+    }
+    auto step_it = _base_to_build_step.find(base_schema->id());
+    if (step_it == _base_to_build_step.end()) {
+        return;// In case all the views for this CF have finished building already.
+    }
+    if (step_it->second.base_schema->version() == base_schema->version()) {
+        return;// Base schema was already updated, probably while creating an index.
+    }
```

Comment:
Why are secondary indexes relevant here? Don't they belong to a higher abstraction layer?

Reply:
They do but they are particularly likely to cause the issues here because they combine base table update and view creation in a single schema update (though it's possible to reach this combination in other ways as well)